### PR TITLE
Feat/pooja/registrationpage

### PR DESCRIPTION
**Pull request by P00-JA**
[https://p00-ja.github.io/registrationPAge/](url)

1. responsive registration page
2. if the given mail id and password matches that in local storage it redirects to user dashboard
3. if  email and password not found gives alert message "no user found"
4. signup creates a user object in local storage if validation successful.

### CODE REVIEW

File: dashBoard.css [hunk 1]
```diff
@@ -0,0 +1,99 @@
+@import url('https://fonts.googleapis.com/css2?family=Cinzel&display=swap');
+@import url('https://fonts.googleapis.com/css2?family=Playpen+Sans:wght@300&display=swap');
+*{
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+    font-family: 'Playpen Sans', cursive;
+}
```
Comment (athulya19ratheesh) [inline dashBoard.css:9]: Add a single line break between each blocks.

File: dashBoard.css [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+body{
+    width: 100%;
+    height: 100vh;
+    background: url("assets/mountainColorful.png") center / cover no-repeat;
+
```
Comment (athulya19ratheesh) [inline dashBoard.css]: Remove unwanted line breaks inside blocks

File: loginPage.css [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    padding: 20px;
+    background: rgba(255, 252, 252, 0.7);
+    border: 3px solid transparent;
+    border-image: linear-gradient(45deg, rgb(255, 0, 132) , rgb(187, 0, 255));
+    border-image-slice: 1;
+    transform: translate(-50%, -60%);
+    max-width: 300px;
+    width: 100%;
+    height: 10rem;
+    border-radius: 15px;
```
Comment (athulya19ratheesh) [inline loginPage.css]: Duplicate css with alertPopup class. reuse the same and only specify function specific additional CSS to separate class.

File: loginPage.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+const nameSignUp=document.getElementById("nameSignup");
+const mobileNumSignUp=document.getElementById("mobileNumSignUp");
+const mailIdSignUp=document.getElementById("mailIdSignUp");
+const passwordSignUp=document.getElementById("passwordSignUp");
+
+//taking values from login form
+const mailIdLogin=document.getElementById("mailIdLogin");
+const passwordLogin=document.getElementById("passwordLogin");
+let storeUserData=[];
+
```
Comment (athulya19ratheesh) [inline loginPage.js]: Page formatting issues

File: loginPage.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+      let decipher = CryptoJS.AES.decrypt(cipher, crypt.secret);
+      return decipher.toString(CryptoJS.enc.Utf8);
+    }
+  };
+//when the login popup button clicked
+loginPopUpButton.addEventListener('click',()=>{
+   document.querySelector(".backgroundBlur").style.display="block"
+   loginPopUpForm.style.display="block";
+})
+//validate login
```
Comment (athulya19ratheesh) [inline loginPage.js:59]: Add a line breaks between functions

File: loginPage.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+        mobileNumSignUp.style.border="3px solid red";
+        alert("Mobile number should'nt contain non numeric values")
+      }else if(!testEmail){
+        mailIdSignUp.style.border="3px solid red";
+        alert("valid email address follows the format: username@domain.com")
+      }else if(!testPassword){
+        passwordSignUp.style.border="3px solid red";
+        alert("At least of 8 Character passwords & a mix of uppercase letters, lowercase letters, numbers, and symbols.")
+      }
+
```
Comment (athulya19ratheesh) [inline loginPage.js]: remove unwanted linebreaks

File: loginPage.css (hunk 3 above)
Comment (P00-JA) [reply loginPage.css]: tried giving same div to both alert but showing alertPopUp null .so resumed to previous method

File: dashBoard.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+const logoutButton=document.getElementById("logoutPopUpButton");
+
```
Comment (athulya19ratheesh) [inline dashBoard.js:2]: Formatting issues

File: index.html [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    <header>
+        <nav class="navBar">
+            <a href="#" class="logo">
+                <img src="assets/hiker.png" alt="logo">
+                <h3>Hikers Point</h3>
+            </a>
+            <button class="loginButton" id="loginPopUpButton">Login</button>
+        </nav>
+    </header>
+    <div class="backgroundBlur"></div>
```
Comment (athulya19ratheesh) [inline index.html:24]: naming?

File: loginPage.css [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+    background: linear-gradient(45deg, rgb(255, 0, 132) , rgb(187, 0, 255));
+    transition: 0.2s ease-in-out;
+}
+
+.signUpPopup button:hover{
+    background: #2e2d2e;
+    border: 3px solid transparent;
+    border-image: linear-gradient(45deg, rgb(255, 0, 132) , rgb(187, 0, 255));
+    border-image-slice: 1;
+}
```
Comment (athulya19ratheesh) [inline loginPage.css]: Duplicate with .popupForm button

File: loginPage.js [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+         resetLogin();
+    }
+});
+
+//validate login
+function loginValidate(){
+    let testEmail=/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g.test(mailIdLogin.value);
+    let testPassword=/^(?=.{8,})(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\s).*$/gm.test(passwordLogin.value);
+    if(testEmail && testPassword){
+        let userInfo=storeUserData.find((user)=>user.mailId==mailIdLogin.value);
```
Comment (athulya19ratheesh) [inline loginPage.js]: use const if not modifying

File: loginPage.js [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+        if (userInfo) {
+            // Decrypt the password
+            let decryptedPassword = crypt.decrypt(userInfo.password);
+            if(!decryptedPassword==passwordLogin.value){
+                alert("Incorrect password!")
```
Comment (athulya19ratheesh) [inline loginPage.js]: formating issues with page

File: loginPage.js [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+            alert("User not found");
+        }
+    }else{
+        alert("Invalid Input!")
+    }
+}
+
+//set cookie
+function setCookie(userName,userValue,expDays){
+    const d=new Date();
```
Comment (athulya19ratheesh) [inline loginPage.js]: naming

File: loginPage.js [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+    d.setTime(d.getTime()+ (expDays*24*60*60*1000));
+    let expiry="expires="+d.toUTCString();
```
Comment (athulya19ratheesh) [inline loginPage.js]: use const

File: loginPage.js [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+  mobile:mobileNumSignUp.value,
+  mailId:mailIdSignUp.value,
+  password:crypt.encrypt(passwordSignUp.value)
+ }
+    storeUserData.push(UserData);
+    saveUserData();
+    resetLogin();
+    resetSignUp();
+    closeSignup();
+
```
Comment (athulya19ratheesh) [inline loginPage.js]: remove unwanted linebreaks inside the blocks

**Review by athulya19ratheesh — CHANGES_REQUESTED** [review]

File: index.html (hunk 8 above)
Comment (P00-JA) [reply index.html:24]: can you mention the issue regarding naming

Comment (athulya19ratheesh) [reply index.html:24]: backgroundBlur stands for?

File: loginPage.js (hunk 12 above)
Comment (P00-JA) [reply loginPage.js]: should i correct it to 
cost date=new Date();

Comment (athulya19ratheesh) [reply loginPage.js]: yes

File: index.html (hunk 8 above)
Comment (P00-JA) [reply index.html:24]: baground blur effect to create a transparent div above when form popsup

File: loginPage.js (hunk 12 above)
Comment (P00-JA) [reply loginPage.js]: done

**Pull request merged**